upup/pkg/fi/cloudup/scalewaytasks/instance.go

```diff
@@ -359,3 +426,54 @@ func (_ *Instance) RenderTerraform(t *terraform.TerraformTarget, actual, expecte
 	}
 	return t.RenderResource("scaleway_instance_server", tfName, tfInstance)
 }
+
+func checkImageDifferences(c *fi.CloudupContext, cloud scaleway.ScwCloud, actualServer *instance.Server, expectedImage string) (bool, error) {
+	localImage, err := cloud.MarketplaceService().GetLocalImageByLabel(&marketplace.GetLocalImageByLabelRequest{
+		ImageLabel:     expectedImage,
+		Zone:           actualServer.Zone,
+		CommercialType: actualServer.CommercialType,
+	}, scw.WithContext(c.Context()))
+	if err != nil {
+		return false, fmt.Errorf("getting image from the marketplace: %w", err)
+	}
+
+	if actualServer.Image.ID != localImage.ID {
+		return true, nil
+	}
+	return false, nil
+}
+
+func checkUserDataDifferences(c *fi.CloudupContext, cloud scaleway.ScwCloud, actualServer *instance.Server, expectedUserData *fi.Resource) (bool, error) {
+	actualUserData, err := cloud.InstanceService().GetServerUserData(&instance.GetServerUserDataRequest{
+		Zone:     actualServer.Zone,
+		ServerID: actualServer.ID,
+		Key:      "cloud-init",
+	}, scw.WithContext(c.Context()))
+	if err != nil {
+		return false, fmt.Errorf("error getting actual user-data: %w", err)
```

Comment:
Error messages shouldn't start with the word "error", as they're printed with an "error: " prefix.